### PR TITLE
radare2: Add support for darwin platforms

### PR DESCRIPTION
**Pull request by arkivm**
###### Motivation for this change
* Add support for darwin
* For `libewf` the configure script tries `AC_CHECK_HEADERS` for `bzlib.h`. On darwin platforms, this header file comes along with the SDK `/Library/Developer/CommandLineTools/SDKs/MacOSX12.0.sdk/usr/include/bzlib.h` and this adds `-lbz2` while linking. Maybe this should be fixed in a totally different way.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by arkivm** [comment]
tagging maintainers @raskin @makefu @mic92

File: pkgs/development/libraries/libewf/default.nix [hunk 1]
```diff
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ zlib openssl libuuid ];
+  buildInputs = [ zlib openssl libuuid ] ++ lib.optionals stdenv.isDarwin [ bzip2 ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libewf/default.nix:13]:
```suggestion
  buildInputs = [ zlib openssl libuuid ]
    ++ lib.optionals stdenv.isDarwin [ bzip2 ];
```

File: pkgs/development/tools/analysis/radare2/default.nix [hunk 2]
```diff
@@ -51,6 +51,13 @@ stdenv.mkDerivation rec {
     chmod -R +w libr/asm/arch/arm/v35arm64/arch-arm64
   '';
 
+  postFixup = lib.optionalString stdenv.isDarwin ''
+    install_name_tool -change libcapstone.4.dylib ${capstone}/lib/libcapstone.4.dylib $out/bin/rasm2
+    install_name_tool -change libcapstone.4.dylib ${capstone}/lib/libcapstone.4.dylib $out/bin/ragg2
+    install_name_tool -change libcapstone.4.dylib ${capstone}/lib/libcapstone.4.dylib $out/bin/rabin2
+    install_name_tool -change libcapstone.4.dylib ${capstone}/lib/libcapstone.4.dylib $out/lib/libr_asm.5.4.2.dylib
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/radare2/default.nix:58]:
```suggestion
    for file in $out/bin/rasm2 $out/bin/ragg2 $out/bin/rabin2 $out/lib/libr_asm.5.4.2.dylib; do
      install_name_tool -change libcapstone.4.dylib ${capstone}/lib/libcapstone.4.dylib $file
    done
```

File: pkgs/development/tools/analysis/radare2/default.nix [hunk 3]
```diff
@@ -59,7 +66,11 @@ stdenv.mkDerivation rec {
   makeFlags = [
     "GITTAP=${version}"
     "RANLIB=${stdenv.cc.bintools.bintools}/bin/${stdenv.cc.bintools.targetPrefix}ranlib"
+  ] ++ lib.optionals stdenv.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/radare2/default.nix:69]:
```suggestion
```

Does this break linux? If not I think we should do it like this to make linux and darwin similar.

File: pkgs/development/tools/analysis/radare2/default.nix [hunk 4]
```diff
@@ -94,11 +105,11 @@ stdenv.mkDerivation rec {
     xxHash
   ];
 
-  meta = {
+  meta = with lib; {
     description = "unix-like reverse engineering framework and commandline tools";
     homepage = "http://radare.org/";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/radare2/default.nix:110]:
```suggestion
    homepage = "https://radare.org/";
```

File: pkgs/development/tools/analysis/radare2/default.nix [hunk 5]
```diff
@@ -94,11 +105,11 @@ stdenv.mkDerivation rec {
...
-    license = lib.licenses.gpl2Plus;
-    maintainers = with lib.maintainers; [ raskin makefu mic92 ];
-    platforms = with lib.platforms; linux;
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ raskin makefu mic92 arkivm ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/radare2/default.nix:113]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/development/tools/analysis/radare2/default.nix (hunk 3 above)
Comment (Mic92) [reply pkgs/development/tools/analysis/radare2/default.nix:69]: This would potentially also help with cross compiling.

**Comment by arkivm** [comment]
@SuperSandro2000 could this be merged?

**Pull request merged**